### PR TITLE
blog: On the salient features of Winglang

### PR DESCRIPTION
**Pull request by Chriscbr**
TODO: fix links

### CODE REVIEW

**Review by eladb — APPROVED** [review]
This is great!

File: blog/2023-06-22-languages-perspective.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
+---
+title: "On the salient features of Winglang: a personal perspective"
+authors: 
+  - rybickic
+tags: [cloud-oriented programming, language design, compilers, inflight, cloud services, immutability]
+hide_table_of_contents: true
+---
+
+In the ever-evolving realm of programming languages, where academia and industry have continuously pushed the boundaries of software design, a new player has emerged: Wing.
```
Comment (staycoolcall911) [inline blog/2023-06-22-languages-perspective.md:9]: Mixing "Wing" and "Winglang" - I'd stick to Winglang throughout the blog post.

File: blog/2023-06-22-languages-perspective.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+From my perspective, Winglang is a system that aims to bridge the gap between cloud infrastructure configuration and computational logic.
+With its two-phased execution model, Wing encourages developers to think holistically about cloud applications, weaving together the configuration of cloud resources and the code that operates within them.
+It's a language that has its roots in the collective wisdom of the programming community, shaped by feedback and collaboration, drawing inspiration from both academic research and real-world industry practices.
+
+In this blog post, I'll share with you the most salient features of Wing that have sparked my enthusiasm so far.
+From the concept of cloud services as first-class citizens to its design principle of immutability by default, Wing has introduced both novel and familiar approaches to enhance developer productivity and mitigate common pitfalls in large-scale systems.
+
+Now, before I dive into the features that make Wing distinctive, it's important to note that the language is still in its alpha stage.
```
Comment (staycoolcall911) [inline blog/2023-06-22-languages-perspective.md:19]: Let's use a different term in place of "Alpha". Maybe "early stage"?

File: blog/2023-06-22-languages-perspective.md [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+## Cloud services as first-class citizens
+
+Wing models the notion of cloud resources and services through a class-based data model where all class objects are named and organized into an in-memory tree. This ensures every object can be assigned an application-unique address or "path" that is deterministic across compilations - a common prerequisite for generating reusable cloud infrastructure configuration. Polymorphic classes can be written which are concretized into different implementations depending on the target cloud platform.*
```
Comment (staycoolcall911) [inline blog/2023-06-22-languages-perspective.md:35]: A good place to use the term "dependency injection". Maybe:
```suggestion
Wing models the notion of cloud resources and services through a class-based data model where all class objects are named and organized into an in-memory tree. This ensures every object can be assigned an application-unique address or "path" that is deterministic across compilations - a common prerequisite for generating reusable cloud infrastructure configuration. Polymorphic classes can be written which are concretized into different implementations using dependency injection based on the target cloud platform.*
```

File: blog/2023-06-22-languages-perspective.md [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+[npm]: https://www.npmjs.com/
+
+## JSON as a first-class type
+
+JSON is one of the most widely used protocols for sending and receiving information between internet services due to its simplicity and human-readability. As the lingua franca of the cloud, Wing includes it as a built-in data type, with `Json` and `MutJson` variants. Some of the features enabled by the JSON type to make life easier for developers are schema validation and the ability to parse JSON values into well-typed structs with compiler support.*
+
+## Local type inference
+
+To save some quantity of programmer key-pressing, Wing supports local type inference: signatures of functions and classes always require type annotation, but within the body of a function, many variables can be declared without any annotations, and Wing will infer the variable’s type from its initialization or its usage.
```
Comment (staycoolcall911) [inline blog/2023-06-22-languages-perspective.md:61]: Is it indeed "initialization or its usage" or just "initialization"? Just couldn't quickly think of an example

File: blog/2023-06-22-languages-perspective.md [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+
+Wing contains the usual assortment of algebraic types from functional languages, like function types, tuples*, disjoint union types*, and structs. Algebraic types can be easily destructured into their components, and can be pattern-matched using the switch/match keyword.*
+
+## Generic code*
+
+Wing supports a number of forms of parametric polymorphism, such as classes, iterators, and functions parameterized by other types.
+
+---
+
+If you're curious to learn more or to see examples of real Wing code, check out the [Winglang website](https://winglang.io/) and its [documentation](https://docs.winglang.io/), and join the [Wing Slack](https://t.winglang.io/slack) to get involved in the community.
```
Comment (staycoolcall911) [inline blog/2023-06-22-languages-perspective.md:81]:
```suggestion
If you're curious to learn more or to see examples of real Wing code, check out the [Winglang website](https://winglang.io/) and its [documentation](https://winglang.io/docs), and join the [Wing Slack](https://t.winglang.io/slack) to get involved in the community.
```

**Review by staycoolcall911 — APPROVED** [review]
Chris - this is great! Beautifully written and interesting. Left a few minor comments

**Pull request closed without merging**